modules/monitoring-attaching-additional-labels-to-your-time-series-and-alerts.adoc

```diff
@@ -0,0 +1,137 @@
+:_module-type: PROCEDURE
+
+// Module included in the following assemblies:
+//
+// * assemblies/configuring-the-monitoring-stack.adoc
+
+[id="attaching-additional-labels-to-your-time-series-and-alerts_{context}"]
+= Attaching additional labels to your time series and alerts
+
+[role="_abstract"]
+Using the external labels feature of Prometheus, you can attach custom labels to all time series and alerts leaving Prometheus.
+
+.Prerequisites
+
+* *If you are configuring core {product-title} monitoring components*:
+** You have access to the cluster as a user with the `cluster-admin` role.
+** You have created the `cluster-monitoring-config` `ConfigMap` object.
```

Comment:
Not applicable for OSD/ROSA

Reply:
Removed.

Reply:
@fahlmant  It also seems like this section should be removed:

** *To attach custom labels to all time series and alerts leaving the Prometheus instance that monitors core {product-title} projects*:
.. Edit the `cluster-monitoring-config` `ConfigMap` object in the `openshift-monitoring` project:
+
[source,terminal]
----
$ oc -n openshift-monitoring edit configmap cluster-monitoring-config
----

.. Define a map of labels you want to add for every metric under `data/config.yaml`:
+
[source,yaml]
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: cluster-monitoring-config
  namespace: openshift-monitoring
data:
  config.yaml: |
    prometheusK8s:
      externalLabels:
        <key>: <value> <1>
----
+
<1> Substitute `<key>: <value>` with a map of key-value pairs where `<key>` is a unique name for the new label and `<value>` is its value.
+
[WARNING]
====
Do not use `prometheus` or `prometheus_replica` as key names, because they are reserved and will be overwritten.
====
+
For example, to add metadata about the region and environment to all time series and alerts, use:
+
[source,yaml]
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: cluster-monitoring-config
  namespace: openshift-monitoring
data:
  config.yaml: |
    prometheusK8s:
      externalLabels:
        region: eu
        environment: prod
----